blog/draft/overweeg-node-modules-in-te-checken.md

```diff
@@ -0,0 +1,148 @@
+Er zijn heel veel packages op de NPM registry met een groot scala aan functionaliteit en diepgang die ze bieden.
+Dit is inherent aan de JavaScript community, waar de programmeertaal van nature weinig "out-of-the-box" aanleverde.
+Er is actief gepromoot om kleine packages te publiceren die een oplossing bieden voor een specifiek probleem.
+
+Over tijd zagen we een trend: de `node_modules` folder in een regulier frontend project kan groot worden als gevolg van een gigantische dependency tree.
+Dit omdat er vaak lagen op lagen van dependencies worden gebruikt om zo tot een groter geheel te komen.
+Het is daarom gebruikelijk om Gigabytes aan disk space kwijt te zijn aan de `node_modules` van alle projecten waar je aan werkt.
+Hier zijn al vaker memes van gemaakt, dus deze blog post zal over deze trend niet verder uitwijden.
+
+![](https://i.redd.it/tfugj4n3l6ez.png)
+
+*Meme van de grootte van een `node_modules` folder. Source: /r/programmerhumor on Reddit*
+
+## Een grote transitive dependency tree zorgt voor complexiteit
+
+Als we naar aspecten als security en dependency management kijken, dan is dat een grote opgave voor frontend projecten.
+De hoeveelheid transitive dependencies die (al dan niet onbewust) in een software applicatie terecht komen is dusdanig groot, dat analyse van al die packages lastig is.
+Tegelijkertijd is de kans op "dependency rot" ook groter naarmate meer dependencies worden meegenomen.
+De kans dat 1 van alle packages in een `node_modules` in het afgelopen jaar niets aan is veranderd is groot.
+Dit zorgt ervoor dat als er een transitive dependency in een stilstaande dependency zit, en deze package niet wordt geupdate, het ook in alle `node_modules` van de afnemers zit.
+
+Het is lastig om dit overzichtelijk te houden, wat in de praktijk resulteert in het vergeten of actief ontwijken van updaten van problematische dependencies.
+En dat is ook menselijk, want het kost simpelweg veel tijd en energie om hier constant bovenop te zitten.
+Als de problemen dan een dermate grootte hebben dat het niet bij te benen is, dan haakt men af.
+Dit zorgt dus inherent voor dat er meer stilstaande packages komen en het probleem verergert.
+
+## Stilstand en tech debt
+
+Deze stilstaande packages en resulterende tech debt zullen in elk project voorkomen, daar is ook niet per se iets verkeerds aan.
+Er moeten prioriteiten worden gesteld en met een naderende deadline zullen er scherpe keuzes worden gemaakt.
+Het schoonhouden van een dependency tree verwacht ik niet van dat het hoog op het lijstje staat.
+
+Hier is wel een verschil in hoe in een private context en in de publieke sector mee om wordt gegaan.
+Waar in een private context er gekeken wordt of een product nog levensvatbaar is, heeft een publieke sector projecten die een stuk langer lopen uit (wettelijke) noodzaak.
+Als er een paar jaar later er dan vulnerabilities ergens in een transitive dependency tree zitten, kan dat lastiger worden voor deze langlopende projecten.

```

Comment:
Misschien hier onderstrepen dat het dus nog belangrijker is om kritische naar je deps te kijken?